examples/example_bids_datasets.py

```diff
@@ -0,0 +1,187 @@
+        # Make new df
+        new_design = pd.DataFrame(columns = ['TR', 'CSplusshock','CSplus','CSminus'])
+        new_design['TR'] = [i for i in range(1, (total_dur+1))]
+        # For each stimulus type, mark the onset of the stimulus with 1. All other values should be 0
+        for cs in [CSplus_shock, CSplus, CSminus]:
+            for i in range(0, len(cs.index)):
+                tr_start = int(round(cs.iloc[i,0]))
+                if cs is CSplus_shock:
+                    new_design.iloc[tr_start, :] =  [tr_start, 1, 0, 0]
+                elif cs is CSplus:
+                    new_design.iloc[tr_start, :] =  [tr_start, 0, 1, 0]
+                elif cs is CSminus:
+                    new_design.iloc[tr_start, :] =  [tr_start, 0, 0, 1]
+                else:
+                    new_design.iloc[tr_start, :] =  [tr_start, 0, 0, 0]
+        # Fill NaNs with 0s
+        new_design.fillna(0, inplace=True)
+        # Give the event file a new name
+        new_name = raw.parse_file_entities(eventfile)
+        new_name['suffix'] = 'events_glmsingle'
+        pattern = 'sub-{}_ses-{}_task-{}_run-{}_{}.csv'
+        # Save the event file in derivatives, exclude TR column
+        new_design.to_csv(os.path.join(root, 'derivatives', 'iter4',
+        'sub-'+new_name['subject'], 'ses-'+new_name['session'], 'func',
+        pattern.format(new_name['subject'],
+        new_name['session'], new_name['task'], new_name['run'],
+        new_name['suffix'])), columns=['CSplusshock','CSplus','CSminus'],
+        index=False, header=False)
+        print('glmsingle event file saved:', new_name['subject'], 'run:', new_name['run'])
+def upsample_bold(processed, tr, tr_new, num_slices, subjects):
+    ''' Upsample task-based fMRI data to 1s using pyslicetime scripts:
+    https://github.com/Charestlab/pyslicetime/blob/master/example_nipype.py.
+    Output will have suffix _stc appended.'''
+    from slicetime.nipype_interface import SliceTime
+    # Grab all preprocessed runs
+    bold_runs = processed.get(subject=subjects, extension='.nii.gz', suffix='bold', return_type='filename')
+    # Filter for resolution - here we exclude res-2
+    bold_runsf = [b for b in bold_runs if 'res-2' not in b]
+    # Import parameters
+    tr_old = tr
+    tr_new = tr_new
+    num_slices = num_slices
+    # sliceorder needs to be 1-based (see fakeout below)
+    slicetimes = np.flip(np.arange(0, tr_old, tr_old/num_slices, dtype=object)).tolist()
+    # Iter over runs
+    for run in bold_runsf:
+        print('Upsampling:', run)
+        st = SliceTime()
+        st.inputs.in_file = run
+        st.inputs.tr_old = tr
+        st.inputs.tr_new = tr_new
+        st.inputs.slicetimes = slicetimes
+        res = st.run()
+
+def smooth_bold(processed, smooth_num, subjects):
+    ''' Smooth data prior to first-level analysis using FSL in nipype.
+    Output will have suffix _smooth appended.'''
+    from nipype.interfaces.fsl import Smooth
```

Comment:
maybe I missed it but you did not import nipype, right? 
and I don't think it is a dependency of glmsingle so this should probably go into the requirements

also not a python expert, but do we have to go through nipype FSL interface to smooth the data? Doesn't that pre supposes that people have FSL installed on their system?